rpmlb/recipe.py

```diff
@@ -138,3 +252,13 @@ def make_sequence(name: str, count: int):
         sequence_map = dict(starmap(make_sequence, count_map.items()))
 
         return sequence_map
+
+
+class RecipeError(Exception):
```

Comment:
Hint: If you want this to be user-faced, nice formatted error message, you can subclass `click.ClickException` and customize its parameters/`format_message()` method.

http://click.pocoo.org/6/exceptions/
https://github.com/pallets/click/blob/master/click/exceptions.py

Example:

```python
class RecipeError(click.ClickException):
    exit_code = 1  # default, can be omitted

    def format_message(self):
      report = 'Recipe file is invalid! Problems found:'

      return '\n\t'.join([report] + self.messages)
```